mlflow/models/evaluation/default_evaluator.py

```diff
@@ -466,14 +468,16 @@ def _evaluate_custom_metric(custom_metric_tuple, eval_df, builtin_metrics):
                                 ``custom_metrics`` parameter of ``mlflow.evaluate``
     :param eval_df: A Pandas dataframe object containing a prediction and a target column.
     :param builtin_metrics: A dictionary of metrics produced by the default evaluator.
+    :param additional_df: A Pandas dataframe that contains additional information for the overall dataset
+    :param additional_array: A numpy array that contains additional information for the overall dataset
```

Comment:
yes, it will look something like this: https://github.com/mlflow/mlflow/issues/8060#issuecomment-1475747487
the indexes on `eval_df` needs to be the same as `additional_df` for it to be concatenated.

i thought that having optional `pd.DataFrame` and `np.array` should be easier rather than accepting `pd.DataFrame` and having to convert to a `np.array` later?

Reply:
What if making `eval_df` containing all columns of `additional_df` and removing `additional_df` argument ? This makes the interface simpler

Reply:
i suppose so. but we will still need to pass in new function parameters for `_evaluate_custom_metrics_and_log_produced_artifacts` to have the additional columns.. which would be a breaking change too?

it seems like we have to create a third party library custom metrics to have this working? because we will have to change the function parameters in either `_generate_model_predictions` or `_evaluate_custom_metrics_and_log_produced_artifacts`. or am i completely off?